executor/runtime/types/container.go

```diff
@@ -1225,7 +1225,14 @@ func populateContainerEnv(c Container, config config.Config, userEnv map[string]
 	}
 	vpcAllocation := c.VPCAllocation()
 	if a := vpcAllocation.IPV4Address(); a != nil {
-		env[metadataserverTypes.EC2IPv4EnvVarName] = a.Address.Address
+		if c.EffectiveNetworkMode() == titus.NetworkConfiguration_Ipv6AndIpv4Fallback.String() {
```

Comment:
Does this cover both IPv6-only and IPv6WithTransition?

Reply:
No, only transition.

Are you sure with *true* v6only containers we *still* want this variable to be set with a dummy value?

I assumed that in true v6 only mode we wouldn't need any workarounds... idk because it doesn't have any fallback mechanism at all for v4?

Reply:
> Are you sure with true v6only containers we still want this variable to be set with a dummy value?

Leaning yes at this point, it would be easier to add a fake value and test different types of services' behavior when it's unset than the other way around.